docs/specs/rate-limit.md

```diff
@@ -63,4 +64,140 @@ Rate limits without default are hard-coded (non-configurable).
 ## Future Works
 
 - We may want to apply request-level rate limits (e.g. admin API, OIDC endpoints)
-- We may want to exclude certain users (e.g. by IP) from applying rate limit.
\ No newline at end of file
+- We may want to exclude certain users (e.g. by IP) from applying rate limit.
+
+
+## Configuration
+
+In general, rate limits are configured using two fields:
+```yaml
+verification:
+    rate_limits:
+        # No rate limit is applied
+        # validate_code_per_ip:
+        #   period: 1h  # required
+        #   brust: 1    # default to 1
+---
+verification:
+    rate_limits:
+        # 1 validation attempt allowed per hour.
+        validate_code_per_ip:
```

Comment:
Do we still have a enabled flag? All pointer to struct in the config will be created recursively. Not sure if we want to give a meaning on the zero value of the rate limit struct.

Reply:
There is no enabled flag in the planned config, there are 3 cases:
- the field is not present (i.e. value is missing): the rate limit config is initialized to zero value, which represents no rate limit is configured.
- the field is set to an object with just `period` field: `brust` is default to 1.
- the field is set to an object with both `period` & `brust` fields.

Setting the field to an object with just `brust` field is an error (required).

Reply:
Looks like we can't do it this way: required field nested in optional field value does not roundtrip correctly.

Reply:
So what is our plan?

Reply:
I suppose we need to keep the enabled flag for simplicity. So the cases are:
```yaml
# rate limit disabled by omitting the field
# validate_code_per_ip:
---
# rate limit disabled by settings enabled
validate_code_per_ip:
    enabled: false
---
# rate limit enabled with default brust (1)
validate_code_per_ip:
    enabled: true
    period: 1h
---
# rate limit enabled
validate_code_per_ip:
    enabled: true
    period: 1h
    brust: 10
```

`period` is required if `enabled` is true.